internal/clients/ec2/aws_supported_types.go

```diff
@@ -10,19 +10,24 @@ import (
 
 func NewInstanceTypes(ctx context.Context, types []types.InstanceTypeInfo) (*[]clients.InstanceType, error) {
 	list := make([]clients.InstanceType, 0, len(types))
+	var storage int64 = 0
 	for i := range types {
 		architectures := types[i].ProcessorInfo.SupportedArchitectures
 		for _, arch := range architectures {
 			arch, err := clients.MapArchitectures(ctx, string(arch))
 			if err != nil {
 				return nil, payloads.ClientError(ctx, "Instance type", "", err, 500)
 			}
+			if types[i].InstanceStorageInfo != nil {
+				storage = *types[i].InstanceStorageInfo.TotalSizeInGB
+			}
 			list = append(list, clients.InstanceType{
 				Name:         string(types[i].InstanceType),
 				VCPUs:        types[i].VCpuInfo.DefaultVCpus,
 				Cores:        types[i].VCpuInfo.DefaultCores,
 				MemoryMiB:    *types[i].MemoryInfo.SizeInMiB,
 				Architecture: string(arch),
+				Storage:      storage,
 				Supported:    clients.IsSupported(string(types[i].InstanceType)),
 			})
 		}
```

Comment:
There is a bug in this code, I suggest you to avoid variable completely, just set the storage flag before appending to list via temporary local one for the `InstanceType` type.

Reply:
Thanks @lzap didn't notice :)